libcudacxx/include/cuda/std/__type_traits/type_list.h

```diff
@@ -58,6 +67,49 @@ struct __type_quote
   using __apply = _Fn<_Ts...>;
 };
 
+//! @brief Turns a unary class or alias template into a meta-callable
+template <template <class> class _Fn>
+struct __type_quote1
```

Comment:
I am non-native speaker, but I cannot guess what the "quote" here means

Reply:
It's "escaping" a type function of the form `F<Ts...>`, which is a template, into a type `Q` that has a nested `::__apply<Ts...>` equivalent to `F<Ts...>`. Quoting converts a meta function (a template) into a type.

See also the Definition section in Boost.Mp11's documentation: https://www.boost.org/doc/libs/1_86_0/libs/mp11/doc/html/mp11.html, which has the same concept, but uses a nested `fn` instead of `__apply`.

Reply:
the name `quote` comes from Lisp, where it is a function to turn code into data. if you squint, that's what `__type_quote` is doing.

https://lisp-docs.github.io/cl-language-reference/chap-3/d-i-dictionary/quote_special-operator